iceoryx_posh/include/iceoryx_posh/internal/popo/typed_subscriber.inl

```diff
@@ -26,6 +26,19 @@ inline TypedSubscriber<T, base_subscriber_t>::TypedSubscriber(const capro::Servi
 {
 }
 
+template <typename T, template <typename, typename, typename> class base_subscriber_t>
+inline cxx::expected<Sample<const T>, ChunkReceiveResult> TypedSubscriber<T, base_subscriber_t>::take_1_0() noexcept
+{
+    auto result = BaseSubscriber::takeChunk();
+    if (result.has_error())
+    {
+        return cxx::error<ChunkReceiveResult>(result.get_error());
+    }
+    auto payloadPtr = reinterpret_cast<T*>(result.value()->payload());
+    auto samplePtr = cxx::unique_ptr<T>(reinterpret_cast<T*>(payloadPtr), BaseSubscriber::m_sampleDeleter);
```

Comment:
you should be able to achieve the same with one `static_cast<T*>`

Reply:
Yes. It is just a pointer. I prefer reinterpret_casts when casting void pointers to typed pointers, it reminds me that something low level is going on.

Reply:
I was just asking because some guidelines restrict the usage of `reinterpret_cast` and then we would need to write exceptions which wouldn't be necessary when using `static_cast`

Reply:
I think I already saw code analyzer tools that complain that `reinterpret_cast` is evil. So if a `static_cast` will do, I would go for this

Reply:
Yes. At some places it will be needed though not here. I find it somewhat hides critical information, it is not comparable to say casting a Derived* to a Base* or a int32_t to an int64_t (which static_cast is mainly used for).